### PR TITLE
Linting, XML to_table and fix JSON viz for XML

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Linting fixes.
- `XML_Document` and `XML_Element` have `to_table` method.
- Add `to_default_visualization_data` to `XML_Document` and `XML_Element`.
- Add support to Table viz to render.

image

image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/XML.enso [hunk 1]
```diff
@@ -673,7 +683,28 @@ type XML_Element
...
+            builder.append ["type", "XML_Element"]
+            builder.append ["tag", self.name]
+            if self.attributes.length > 0 then builder.append ["attributes", self.attributes.to_js_object]
+            if self.children_cache.length > 0 then
+                inner_render = if nested then _.to_display_text else
+                    c-> case c of
+                        _ : XML_Element -> c.to_default_visualization_data nested=True as_js_object=True
+                        _ -> c
+
+                ## Limit to 50 children to avoid excessive rendering.
```
Comment (GregoryTravis) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/XML.enso:703]: Should we add some kind of `...` to show the user that elements beyond 50 were hidden?

**Review by GregoryTravis — APPROVED** [review]

**Review by AdRiley — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/XML.enso [hunk 2]
```diff
@@ -369,6 +369,16 @@ type XML_Document
     to_display_text : Text
     to_display_text self = "XML_Document (" + self.root_element.to_display_text + ")"
 
+    ## PRIVATE
+       Transform the vector into text for displaying as part of its default
+       visualization.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/XML.enso]: This comment feels like it's wrong?

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/XML.enso [hunk 3]
```diff
@@ -369,6 +369,16 @@ type XML_Document
...
+    to_default_visualization_data : Text
+    to_default_visualization_data self =
+        outer_vec = Vector.build initial_capacity=4 builder->
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/XML.enso]:
```suggestion
        outer_vec = Vector.build initial_capacity=2 builder->
```

the vector here contains 2 elements - 2-element-vectors.

**Review by radeusgd — APPROVED** [review]
Looks good. Few small nits.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/XML.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/XML.enso:703]: We have `children_count` telling the true amount, so I imagine it's the responsibility of the visualization to see if `children.length < children_count` and display a `...`.

**Pull request merged**